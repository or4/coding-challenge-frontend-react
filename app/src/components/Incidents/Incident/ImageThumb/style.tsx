import styled, { StyledComponent } from 'styled-components';
import { media } from 'ui/utils';

import prevewSvg from './assets/preview.svg';
import { IProps } from '.';

interface IContainerProps {
    isPreview: boolean;
}

export const Container: StyledComponent<'div', any, IContainerProps, never> = styled.div`
    ${media.mobile`
        display: ${(props: IContainerProps) => (props.isPreview ? 'none' : 'inherit')};
		padding-bottom: 10px;
  	`}

    ${media.tabletNDesktop`
		width: 100px;
		height: 100px;
		min-width: 100px;
		min-height: 100px;

		display: flex;
		align-items: center;
		justify-content: center;
		box-sizing: border-box;
		border-radius: 4px;
		border: 1px solid rgba(0, 0, 0, 0.05);
    	background: rgba(0, 0, 0, 0.02);
	`}
`;

export const Image = styled.div`
    background-image: url(${(props: IProps) => props.imageUrl});
    background-repeat: no-repeat;
    background-size: cover;

    ${media.mobile`
		padding-top: 100%;
		width: 100%;
  	`}

    ${media.tabletNDesktop`
		width: 90px;
		height: 90px;
	`}
`;

export const Preview = styled.div`
    background-image: url(${prevewSvg});
    background-repeat: no-repeat;
    background-size: cover;

    width: 55px;
    height: 32px;

    ${media.mobile`
		display: none;
  	`}
`;
