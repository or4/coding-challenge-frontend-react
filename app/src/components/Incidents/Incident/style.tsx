import styled from 'styled-components';
import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    cursor: pointer;
    padding-top: 10px;
    padding-bottom: 10px;
    font-size: 14px;
    outline: none;

    ${media.mobile`
		flex-direction: column;
		border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  	`}

    ${media.tabletNDesktop`
    	min-height: 100px;
	`}

    &:hover {
        background: rgba(0, 0, 0, 0.03);
    }
`;

export const TextContainer = styled.div`
    display: flex;
    flex-direction: column;

    ${media.tabletNDesktop`
        padding: 0 15px;
    `}
`;

export const IncidentTitle = styled.div`
    font-weight: 600;

    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDescription = styled.div`
    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDetails = styled.div``;
