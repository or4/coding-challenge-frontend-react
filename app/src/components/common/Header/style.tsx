import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    align-items: center;
    cursor: pointer;

    ${media.mobile`
        height: 60px;
        margin-bottom: 15px;
    `}

    ${media.tablet`
        height: 100px;
    `}

    ${media.desktop`
        height: 140px;
    `}
`;

export const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.tabletNDesktop`
        padding: 0 18px;
    `}
`;

export const TextWrapper = styled.div`
    padding-left: 15px;
`;

export const Title = styled.div`
    font-weight: 600;

	${media.mobile`
    	font-size: 18px;
  	`}

	${media.tablet`
    	font-size: 32px;
	`}

    ${media.desktop`
    	font-size: 48px;
	`}
`;

export const Description = styled.div`
	font-weight: 600;

	${media.mobile`
    	font-size: 14px;
  	`}

	${media.tablet`
    	font-size: 16px;
	`}

    ${media.desktop`
    	font-size: 32px;
	`}
`;
