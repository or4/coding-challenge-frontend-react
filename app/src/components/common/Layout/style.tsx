import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    margin: 0 auto 15px;

    ${media.mobile`
		padding: 15px;
  	`}

	${media.tablet`
		padding: 15px;
	`}

    ${media.desktop`
		max-width: 940px;
	`}
`;
