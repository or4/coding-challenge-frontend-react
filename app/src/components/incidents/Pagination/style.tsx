import styled from 'styled-components';

import { PageButton } from 'ui/PageButton';

export const Container = styled.div`
    text-align: center;
    margin: 15px 0;
`;

export const Button = styled(PageButton)`
    margin-right: calc(15px + 3px);

    &:last-child {
        margin-right: 0;
    }
`;
