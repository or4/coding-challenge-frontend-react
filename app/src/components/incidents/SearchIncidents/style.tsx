import styled from 'styled-components';
import { media } from 'ui/utils';

export const QueryInput = styled.input`
    min-width: 240px;
    padding: 4px 8px;
    flex-grow: 1;
    margin: 5px 5px;

    ${media.mobileNTablet`
		width: 100%;
	`}
`;
