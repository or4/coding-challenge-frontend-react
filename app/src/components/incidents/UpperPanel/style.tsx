import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    margin: -5px -5px;
`;
