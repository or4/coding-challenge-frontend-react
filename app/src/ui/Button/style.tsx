import styled from 'styled-components';

import { IProps as IButtonProps } from '.';

export const Container = styled.div`
    display: inline-block;
    box-sizing: border-box;
    color: rgba(0, 0, 0, 0.9);
    background-color: #ffffff;
    cursor: ${(props: IButtonProps) => (props.canClick ? 'pointer' : 'default')};
    box-shadow: 3px 3px 0px rgba(0, 0, 0, 0.9);
    border: 3px solid rgba(0, 0, 0, 0.9);

    line-height: 14px;
    height: 30px;
    user-select: none;

    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 100%;

    font-size: 14px;
    padding: 5px 10px;

    &:hover {
        background-color: rgba(0, 0, 0, 0.05);
    }

    ${(props: IButtonProps) =>
        props.canClick
            ? `
                &:active {
                    background-color: rgba(0, 0, 0, 0.9);
                    box-shadow: 0px 0px 0px rgba(0, 0, 0, 0.9);
                    border: 3px solid transparent;
                    color: #ffffff;
                }`
            : ''}
`;
