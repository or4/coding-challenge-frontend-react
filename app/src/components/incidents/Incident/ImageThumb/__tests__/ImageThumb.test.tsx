import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import 'jest-styled-components';
import { ImageThumb } from '..';
import { Container, Preview, Image } from '../style';

const url = 'https://test-image-url.com';

describe('ImageThumb', () => {
    it('should render correct with url', () => {
        const tree = renderer.create(<ImageThumb imageUrlThumb={url} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should contain image component', () => {
        const wrapper = shallow(<ImageThumb imageUrlThumb={url} />);

        expect(wrapper.find(Image)).toHaveLength(1);
    });

    it('should contain preview component when url is falsy', () => {
        expect(shallow(<ImageThumb imageUrlThumb={''} />).find(Preview)).toHaveLength(1);
        expect(shallow(<ImageThumb />).find(Preview)).toHaveLength(1);
    });

    it('should have container', () => {
        const wrapper = shallow(<ImageThumb imageUrlThumb={url} />);

        expect(wrapper.find(Container)).toHaveLength(1);
    });

    it('should have alt', () => {
        const wrapper = shallow(<ImageThumb imageUrlThumb={url} />);

        expect(wrapper.find('img').prop('alt')).toBe('incident');
    });
});

describe('Preview', () => {
    it('should have background image', () => {
        const tree = renderer.create(<Preview />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('Image', () => {
    it('should pass prop with background image', () => {
        const tree = renderer.create(<Image />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
