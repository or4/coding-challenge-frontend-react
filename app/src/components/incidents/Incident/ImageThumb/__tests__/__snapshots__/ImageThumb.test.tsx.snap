// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should pass prop with background image 1`] = `
.c0 {
  background-repeat: no-repeat;
  background-size: cover;
}

@media (max-width:479px) {
  .c0 {
    padding-top: 100%;
    width: 100%;
  }
}

@media (min-width:480px) {
  .c0 {
    width: 90px;
    height: 90px;
  }
}

<div
  className="c0"
/>
`;

exports[`ImageThumb should render correct with url 1`] = `
.c1 {
  background-repeat: no-repeat;
  background-size: cover;
}

@media (max-width:479px) {
  .c0 {
    display: inherit;
    padding-bottom: 10px;
  }
}

@media (min-width:480px) {
  .c0 {
    width: 100px;
    height: 100px;
    min-width: 100px;
    min-height: 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    box-sizing: border-box;
    border-radius: 4px;
    border: 1px solid rgba(0,0,0,0.05);
    background: rgba(0,0,0,0.02);
  }
}

@media (max-width:479px) {
  .c1 {
    padding-top: 100%;
    width: 100%;
  }
}

@media (min-width:480px) {
  .c1 {
    width: 90px;
    height: 90px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    data-test-id="incident-image__image"
    style={
      Object {
        "backgroundImage": "url(https://test-image-url.com)",
      }
    }
  >
    <img
      alt="incident"
      height="0"
      src="https://test-image-url.com"
    />
  </div>
</div>
`;

exports[`Preview should have background image 1`] = `
.c0 {
  background-image: url(preview.svg);
  background-repeat: no-repeat;
  background-size: cover;
  width: 55px;
  height: 32px;
}

@media (max-width:479px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;
